bin/xcm-tools/src/cli.rs

```diff
@@ -61,38 +57,75 @@ pub struct AssetIdCmd {
     pub asset_id: u128,
 }
 
-/// Helper that prints AccountId32 hash value for the derived multilocation.
+/// Helper that prints the derived AccountId32 value for the multilocation.
 #[derive(Debug, clap::Parser)]
-pub struct Account32HashCmd {
+pub struct RemoteAccountCmd {
     /// Parachain id in case sender is from a sibling parachain.
     #[clap(short, long, default_value = None)]
     pub parachain_id: Option<u32>,
-    /// AccountId32 (SS58 scheme, public key) of the sender account.
-    #[clap(short, long, value_parser = account_id_32_parser)]
-    pub account_id_32: [u8; 32],
-    /// NetworkId of the AccountId32 - if not provided, will be set to `Any`
+    /// Public key (SS58 or H160) in hex format. Must be either 32 or 20 bytes long.
     #[clap(short, long)]
-    pub network_id: Option<String>,
+    pub account_key: AccountWrapper,
+}
+
+#[derive(Debug, Clone, Copy)]
+pub struct AccountWrapper {
+    account: [u8; 32],
+    is_32: bool,
 }
```

Comment:
This should be a tuple enum

that will make make it more clean, and replace the `is_32_bytes()` with idiomatic rust pattern matching.

Reply:
Great comment, thanks!